chat-plugins/poll.js

```diff
@@ -0,0 +1,149 @@
+		});
+		output += '</div>';
+
+		return output;
+	};
+
+	Poll.prototype.generateResults = function () {
+		var output = '<div class="infobox" style="text-align:center"><p style="font-weight:bold;font-size:14pt">' + this.question + '</p>';
+		this.options.forEach(function (option, number) {
+			output += '<span style="font-weight:bold;padding:2px">' + number + '. ' + Tools.escapeHTML(option.name) + '</span><br/><span style="font-size:8pt;padding:0px">' + option.votes + ' votes (' + Math.round((option.votes * 100) / this.totalVotes) + '% of total).</span><br/>';
+		});
+		output += '</div>';
+
+		return output;
+	};
+
+	Poll.prototype.update = function () {
+		var results = this.generateResults();
+
+		// Update the poll results for everyone that has voted
+		for (var i = 0; i < this.room.users.length; i++) {
+			var user = this.room.users[i];
+			if (this.voters.indexOf(user.latestIp) >= 0) {
+				user.sendTo(this.room, '|uhtmlchange|poll|' + results);
+			}
+		}
+	};
+
+	Poll.prototype.display = function (user, broadcast) {
+		var votes = this.generateVotes();
+		var results = this.generateResults();
+
+		var target = [];
+
+		if (broadcast) {
+			target = this.room.users;
+		} else {
+			target.push(user);
+		}
+
+		for (var i = 0; i < target.length; i++) {
+			var thisUser = target[i];
+			if (this.voters.indexOf(thisUser.latestIp) >= 0) {
+				thisUser.sendTo(this.room, '|uhtml|poll|' + results);
+			} else {
+				thisUser.sendTo(this.room, '|uhtml|poll|' + votes);
+			}
+		}
+	};
+
+	Poll.prototype.end = function () {
+		var results = this.generateResults();
+
+		this.room.send('|uhtml|poll|' + results);
+		this.room.send('|uhtmlchange|poll|' + results);
+	};
+
+	return Poll;
+})();
+
+exports.commands = {
+	newpoll: function (target, room, user) {
+		if (!this.can('ban', null, room)) return false;
+		if (room.poll) return this.sendReply("There is already a poll in progress in this room.");
+
+		var targets = target.split(target.includes('|') ? '|' : ',');
+		if (targets.length < 3) {
+			return this.sendReply("Not enough arguments for /newpoll.");
+		}
+
+		var options = [];
+
+		for (var i = 1; i < targets.length; i++) {
+			options.push(targets[i]);
+		}
+
+		if (options.length > 8) {
+			return this.sendReply("Too many options for poll (maximum is 8).");
+		}
+
+		room.poll = new Poll(room, Tools.escapeHTML(targets[0]), options);
+		room.poll.display(user, true);
+	},
+
+	poll: function (target, room, user) {
+		if (!room.poll) return this.sendReply("There is no poll running in this room.");
+		if (!this.canBroadcast()) return;
+
+		room.poll.display(user, this.broadcasting);
+	},
+
+	votepoll: function (target, room, user) {
+		if (!room.poll) return this.sendReply("There is no poll running in this room.");
+		if (!target) return this.sendReply("Please specify an option.");
+
+		var parsed = parseInt(target);
+		if (isNaN(parsed)) return this.sendReply("To vote, specify the NUMBER of the option.");
```

Comment:
I don't think you understand what Slayer95 was saying, because you didn't do what he asked here...